daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/CodeGeneratorState.scala

```diff
@@ -162,10 +322,31 @@ class CodeGeneratorState {
   def addComplexTypeStatements(child: ElementBase): Unit = {
     val C = localName(child)
     val e = child.name
+    val hasChoice = structs.top.initChoiceStatements.nonEmpty
```

Comment:
If I'm reading this right, it seems this only supports direct dispatch choices that are the model group of a complex type, e.g.:
```xml
<xs:element name="foo">
  <xs:complexType>
    <xs:choice dfdl:choiceDispatchKey="...">
      ...
    </xs:choice>
  </xs:complexType>
</xs:element>
```
So sequences with "anonymous" choices are not suppported? I guess supporting those adds more complications since you currently use ``_choice`` to figure out which union member is used, and anonmous choices means you could have multiple unions and _choice's, which someone need a distint name?

Does this distint name issue also arise in sequence with multiple elements with the same name? For example:
```xml
<xs:element name="foo">
  <xs:complexType>
    <xs:sequence>
      <xs:element name="a" ... />
      <xs:element name="b" ... />
      <xs:element name="a" ... />
    </xs:choice>
  </xs:complexType>
</xs:element>
```
Would this currently result in to struct members with the name "a" and cause an error?

Reply:
I think for runtime2 it would be acceptable to rule out this sort of schema - requiring all children to have unique names seems like a sensible restriction given all the other restrictions in runtime2.